helevents/models.py

```diff
@@ -1,6 +1,128 @@
+from functools import reduce
+
+from django.conf import settings
+from django_orghierarchy.models import Organization
 from helusers.models import AbstractUser
 
-from events.permissions import UserModelPermissionMixin
+from events.models import PublicationStatus
+
+
+class UserModelPermissionMixin:
+    """Permission mixin for user models
+
+    A mixin class that provides permission check methods
+    for user models.
+    """
+
+    @property
+    def is_external(self):
+        """Check if the user is an external user"""
+        return (
+            not self.organization_memberships.exists()
+            and not self.admin_organizations.exists()
+        )
```

Comment:
I guess this could be a cached property. 🤔 The performance difference is probably pretty negligible, but it would also prevent the value to be changed during a request, which could lead to hard-to-debug errors. Of course, the probability of that happening is pretty close to 0, but still 😄

I think the other methods, like `get_admin_organizations_and_descendants()` could use memoization as well, but that is not in the scope of this PR.

Reply:
Good point! Yeah, noticed that myself as well that some of the properties/methods/functions might as well be cached... but trying to stick to the scope here, it's way too easy to get sidetracked in this project 😅 

Turned this into a cached property.